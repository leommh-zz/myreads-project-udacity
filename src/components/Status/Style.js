const style = {
    status:{
        backgroundColor: '#c8a3ff',
        padding: 15,
    },
    title: {
        color: '#12071d',
    },
}

export default style;