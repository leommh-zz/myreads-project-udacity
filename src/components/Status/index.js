import React from 'react';
import { FormControl, Typography, Select, MenuItem } from '@material-ui/core';
import Style from './Style';

const Status = ({ defineStatus, book, shelf }) => (
    <FormControl style={Style.status}>
        <Typography variant='subtitle1' style={Style.title}> 
            Status
        </Typography> 
        <Select
            value={shelf || 'none'}
            onChange={(e) => defineStatus(book, e.target.value)}
        >
            <MenuItem value='none'><em>None</em></MenuItem>
            <MenuItem value='currentlyReading'>Currently Reading</MenuItem>
            <MenuItem value='wantToRead'>Want to Read</MenuItem>
            <MenuItem value='read'>Read</MenuItem>
        </Select>
    </FormControl>
)

export default Status;