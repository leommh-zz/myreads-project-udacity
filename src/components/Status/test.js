import React from 'react';
import { shallow } from 'enzyme';
import Status from './index';

describe('[Component] Status', () => {

    const defineStatus = (book, shelf) => console.log('Status Updated');
    const book = {
        id: 1,
        title: "Test Book"
    };
    const shelf = "read";

    const wrapper = shallow(<Status defineStatus={defineStatus} book={book} shelf={shelf} />);

    it('Display Status Component with props', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});