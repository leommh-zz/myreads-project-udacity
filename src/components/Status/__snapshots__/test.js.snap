// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Status Display Status Component with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Status
    book={
        Object {
            "id": 1,
            "title": "Test Book",
          }
    }
    defineStatus={[Function]}
    shelf="read"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          style={
                    Object {
                              "color": "#12071d",
                            }
          }
          variant="subtitle1"
>
          Status
</WithStyles(Typography)>,
        <WithStyles(Select)
          onChange={[Function]}
          value="read"
>
          <WithStyles(MenuItem)
                    value="none"
          >
                    <em>
                              None
                    </em>
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
                    value="currentlyReading"
          >
                    Currently Reading
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
                    value="wantToRead"
          >
                    Want to Read
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
                    value="read"
          >
                    Read
          </WithStyles(MenuItem)>
</WithStyles(Select)>,
      ],
      "style": Object {
        "backgroundColor": "#c8a3ff",
        "padding": 15,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Status",
          "style": Object {
            "color": "#12071d",
          },
          "variant": "subtitle1",
        },
        "ref": null,
        "rendered": "Status",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(MenuItem)
              value="none"
>
              <em>
                            None
              </em>
</WithStyles(MenuItem)>,
            <WithStyles(MenuItem)
              value="currentlyReading"
>
              Currently Reading
</WithStyles(MenuItem)>,
            <WithStyles(MenuItem)
              value="wantToRead"
>
              Want to Read
</WithStyles(MenuItem)>,
            <WithStyles(MenuItem)
              value="read"
>
              Read
</WithStyles(MenuItem)>,
          ],
          "onChange": [Function],
          "value": "read",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <em>
                None
</em>,
              "value": "none",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "None",
              },
              "ref": null,
              "rendered": "None",
              "type": "em",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Currently Reading",
              "value": "currentlyReading",
            },
            "ref": null,
            "rendered": "Currently Reading",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Want to Read",
              "value": "wantToRead",
            },
            "ref": null,
            "rendered": "Want to Read",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Read",
              "value": "read",
            },
            "ref": null,
            "rendered": "Read",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            style={
                        Object {
                                    "color": "#12071d",
                                  }
            }
            variant="subtitle1"
>
            Status
</WithStyles(Typography)>,
          <WithStyles(Select)
            onChange={[Function]}
            value="read"
>
            <WithStyles(MenuItem)
                        value="none"
            >
                        <em>
                                    None
                        </em>
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
                        value="currentlyReading"
            >
                        Currently Reading
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
                        value="wantToRead"
            >
                        Want to Read
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
                        value="read"
            >
                        Read
            </WithStyles(MenuItem)>
</WithStyles(Select)>,
        ],
        "style": Object {
          "backgroundColor": "#c8a3ff",
          "padding": 15,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Status",
            "style": Object {
              "color": "#12071d",
            },
            "variant": "subtitle1",
          },
          "ref": null,
          "rendered": "Status",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(MenuItem)
                value="none"
>
                <em>
                                None
                </em>
</WithStyles(MenuItem)>,
              <WithStyles(MenuItem)
                value="currentlyReading"
>
                Currently Reading
</WithStyles(MenuItem)>,
              <WithStyles(MenuItem)
                value="wantToRead"
>
                Want to Read
</WithStyles(MenuItem)>,
              <WithStyles(MenuItem)
                value="read"
>
                Read
</WithStyles(MenuItem)>,
            ],
            "onChange": [Function],
            "value": "read",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <em>
                  None
</em>,
                "value": "none",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "None",
                },
                "ref": null,
                "rendered": "None",
                "type": "em",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Currently Reading",
                "value": "currentlyReading",
              },
              "ref": null,
              "rendered": "Currently Reading",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Want to Read",
                "value": "wantToRead",
              },
              "ref": null,
              "rendered": "Want to Read",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Read",
                "value": "read",
              },
              "ref": null,
              "rendered": "Read",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
